include/envoy/upstream/upstream.h

```diff
@@ -412,6 +414,22 @@ class Cluster : public virtual HostSet {
    * resolution is complete.
    */
   virtual void setInitializedCb(std::function<void()> callback) PURE;
+
+  /**
+   * @return the HostSet of the primary hosts.
+   *
+   * This is the set of the hosts which should receive all traffic for the
+   * cluster as long as the primary hosts remain healthy.
```

Comment:
Q: How is this going to play with panic threshold? Presumably, it will be up to the individual load balancer, and my guess is that the load balancer would first check for healthy in the main set, then if there is no failover set, do panic threshold for the main set. If there is a failover set, it would use the failover set, and if none are healthy there do panic threshold on the main set? Not really asking for a change here but should probably think about this and maybe update docs around this feature.